pkg/controller/stacks/install/installjob.go

```diff
@@ -70,15 +70,17 @@ type prepareInstallJobParams struct {
 	stackManagerPullPolicy corev1.PullPolicy
 	imagePullPolicy        corev1.PullPolicy
 	labels                 map[string]string
+	annotations            map[string]string
 	imagePullSecrets       []corev1.LocalObjectReference
 }
 
 func prepareInstallJob(p prepareInstallJobParams) *batchv1.Job {
 	return &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      p.name,
-			Namespace: p.namespace,
-			Labels:    p.labels,
+			Name:        p.name,
```

Comment:
the name that will be used on this job also needs to be truncated to label (<=63) length too I think, if I'm reading this error message correctly:

```
message: 'Job.batch "crossplane-system.crossplane-stack-minimal-gcpv0.2.2-14-g96f534a"
        is invalid: spec.template.labels: Invalid value: "crossplane-system.crossplane-stack-minimal-gcpv0.2.2-14-g96f534a":
        must be no more than 63 characters'
```
I'm interpreting that to mean that the name given to the job, `crossplane-system.crossplane-stack-minimal-gcpv0.2.2-14-g96f534a` is being automatically applied as a label in the job's `spec.template.labels`, so it needs to be truncated.

Here's a snippet of the job and its relevant `spec.template.labels` that seems to confirm that the job's name gets inserted as a `job-name` label inside it's `spec.template.labels`:
```
spec:
  template:
    metadata:
      creationTimestamp: null
      labels:
        controller-uid: 643d1d2f-d6c9-4738-bd86-e8a478268146
        job-name: crossplane-system.crossplane-provider-gcpv0.6.0-rc.29.g762643b
```

The full job yaml output for your convenience can be seen in this gist: https://gist.github.com/jbw976/2c9def142b4e24fd1e39b9f3f439856e

Reply:
`prepareInstallJob` is called with the name already truncated:
https://github.com/crossplane/crossplane/pull/1338/files#diff-a2cd537f6531321fbef9a92d96b2c425R351-R367

The truncation happens within `ObjectReferenceOnHost`:

https://github.com/crossplane/crossplane/pull/1338/files#diff-33a4f973d281ef627a1c398d9941fc68R73-R84

Reply:
Ahh, now I see where/how that happens, thanks for pointing that out!